src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentProjectsFinder.cs

```diff
@@ -300,154 +217,75 @@ private static bool IsInternalsVisibleToAttribute(AttributeData attr)
-            if (isSubmission)
+            if (symbolOrigination.sourceProject?.IsSubmission == true)
                 return;
 
-            var internalsVisibleToMap = CreateInternalsVisibleToMap(sourceAssembly);
-
-            var sourceAssemblySymbolKey = sourceAssembly.GetSymbolKey(cancellationToken);
-
-            // TODO(cyrusn): What about error tolerance situations.  Do we maybe want to search
-            // transitive dependencies as well?  Even if the code wouldn't compile, they may be
-            // things we want to find.
-            foreach (var projectId in solution.ProjectIds)
+            // Set of assembly names that `assembly` has IVT to.  Computed on demand once needed.
+            HashSet<string>? internalsVisibleToSet = null;
+            foreach (var project in solution.Projects)
             {
-                var project = solution.GetRequiredProject(projectId);
-
-                cancellationToken.ThrowIfCancellationRequested();
-
-                if (project.SupportsCompilation && HasReferenceTo(sourceAssembly, sourceProject, project, cancellationToken))
+                if (!project.SupportsCompilation ||
+                    !HasReferenceTo(symbolOrigination, project, cancellationToken))
                 {
-                    var hasInternalsAccess = await HasInternalsAccessAsync(
-                        sourceAssembly, internalsVisibleToMap,
-                        sourceAssemblySymbolKey, project, cancellationToken).ConfigureAwait(false);
-
-                    dependentProjects.Add(new DependentProject(project.Id, hasInternalsAccess));
+                    continue;
                 }
-            }
-        }
 
-        private static async Task<bool> HasInternalsAccessAsync(
-            IAssemblySymbol sourceAssembly, Lazy<HashSet<string>> internalsVisibleToMap,
-            SymbolKey sourceAssemblySymbolKey, Project project, CancellationToken cancellationToken)
-        {
-            if (internalsVisibleToMap.Value.Contains(project.AssemblyName) &&
-                project.SupportsCompilation)
-            {
-                var compilation = await project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
-
-                var targetAssembly = compilation.Assembly;
-                if (sourceAssembly.Language != targetAssembly.Language)
-                {
-                    var resolvedSymbol = sourceAssemblySymbolKey.Resolve(compilation, cancellationToken: cancellationToken).Symbol;
-                    if (resolvedSymbol is IAssemblySymbol sourceAssemblyInTargetCompilation)
-                    {
-                        return targetAssembly.IsSameAssemblyOrHasFriendAccessTo(sourceAssemblyInTargetCompilation);
-                    }
-                }
-                else
-                {
-                    return targetAssembly.IsSameAssemblyOrHasFriendAccessTo(sourceAssembly);
-                }
```

Comment:
In the end, the change here is effectively replacing this block with "return true" that just treats this as an approximate algorithm?

Reply:
And I guess replacing the cache pattern to use AsyncLazy to allow parallel computation?

Reply:
yup!

Reply:
Effectively yes :)